example/build_cmake_rdc/functor.cpp

```diff
@@ -0,0 +1,21 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <Kokkos_Macros.hpp>
+
+KOKKOS_EXTERNAL_FUNCTION void count_even(const long i, long& lcount) {
```

Comment:
How does that work for CUDA and HIP?
As I understand it, with the current definition of `KOKKOS_EXTERNAL_FUNCTION`, the `__host__ __device__` annotations are missing.

Reply:
Fixed.